fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/GenericDataServiceImpl.java

```diff
@@ -160,23 +161,32 @@ public String generateJsonFromGenericResultsetData(final GenericResultsetData gr
                 if (currColType == null && colType.equalsIgnoreCase("DATE")) {
                     currColType = "DATE";
                 }
+                if (currColType == null && colType.equalsIgnoreCase("DATETIME")) {
+                    currColType = "DATETIME";
+                }
                 currVal = row.get(j);
                 if (currVal != null && currColType != null) {
                     if (currColType.equals("DECIMAL") || currColType.equals("INTEGER")) {
                         writer.append(currVal);
                     } else {
                         if (currColType.equals("DATE")) {
-                            final LocalDate localDate = LocalDate.parse(currVal);
+                            final LocalDate localDate = LocalDate.parse(currVal.toString());
                             writer.append(
                                     "[" + localDate.getYear() + ", " + localDate.getMonthValue() + ", " + localDate.getDayOfMonth() + "]");
                         } else if (currColType.equals("DATETIME")) {
-                            final LocalDateTime localDateTime = LocalDateTime.parse(formatDateTimeValue(currVal),
-                                    DateUtils.DEFAULT_DATETIME_FORMATER);
-                            writer.append("[" + localDateTime.getYear() + ", " + localDateTime.getMonthValue() + ", "
-                                    + localDateTime.getDayOfMonth() + ", " + localDateTime.getHour() + ", " + localDateTime.getMinute()
-                                    + ", " + localDateTime.getSecond() + ", " + localDateTime.get(ChronoField.MILLI_OF_SECOND) + "]");
+                            Calendar dateVal = Calendar.getInstance();
+                            if (currVal instanceof Date) {
+                                dateVal.setTime((Date) currVal);
+                            } else if (currVal instanceof LocalDateTime) {
+                                dateVal.setTime(
+                                        Date.from(((LocalDateTime) currVal).atZone(DateUtils.getDateTimeZoneOfTenant()).toInstant()));
```

Comment:
You sure about this cause I'm not. Are we ensuring that we're saving the datetime value into the datatable with the tenant's timezone?
I don't think so. Can you please double-check?
cc @adamsaghy

Reply:
MySQL 8 sends this exception in the `rs.getTimeStamp` method, so that was the reason

<img width="1739" alt="Screen Shot 2022-07-24 at 19 42 10" src="https://user-images.githubusercontent.com/44206706/180820330-a6580466-4ca4-4b15-8537-7da8b7313df0.png">

Reply:
I think you misunderstood it Alberto.
The only thing I'm saying is about the TimeZone.
If we're saving the datetime value into the DB with TimeZoneX and reading it back and converting it to TimeZoneY, that's definitely not good cause we're messing up the offsets.

Reply:
Done! It was removed due we're using LocalDateTime now